app/ruffle/src/main/java/rs/ruffle/MainActivity.java

```diff
@@ -89,4 +90,93 @@ protected void onCreate(Bundle savedInstanceState) {
+//                return true;
+//            }
+//            if (key_event.getKeyCode() == KeyEvent.KEYCODE_DPAD_UP) {
+//                SimulateKeyDown(KeyEvent.KEYCODE_W);
+//                return true;
+//            }
+//            if (key_event.getKeyCode() == KeyEvent.KEYCODE_DPAD_DOWN) {
+//                SimulateKeyDown(KeyEvent.KEYCODE_S);
+//                return true;
+//            }
+//            if (key_event.getKeyCode() == KeyEvent.KEYCODE_DPAD_LEFT) {
+//                SimulateKeyDown(KeyEvent.KEYCODE_A);
+//                return true;
+//            }
+//            if (key_event.getKeyCode() == KeyEvent.KEYCODE_DPAD_RIGHT) {
+//                SimulateKeyDown(KeyEvent.KEYCODE_D);
+//                return true;
+//            }
+//        }
+//        if (key_event.getAction() == KeyEvent.ACTION_UP) {
+//            if (key_event.getKeyCode() == KeyEvent.KEYCODE_DPAD_UP) {
+//                SimulateKeyUp(KeyEvent.KEYCODE_W);
+//                return true;
+//            }
+//            if (key_event.getKeyCode() == KeyEvent.KEYCODE_DPAD_DOWN) {
+//                SimulateKeyUp(KeyEvent.KEYCODE_S);
+//                return true;
+//            }
+//            if (key_event.getKeyCode() == KeyEvent.KEYCODE_DPAD_LEFT) {
+//                SimulateKeyUp(KeyEvent.KEYCODE_A);
+//                return true;
+//            }
+//            if (key_event.getKeyCode() == KeyEvent.KEYCODE_DPAD_RIGHT) {
+//                SimulateKeyUp(KeyEvent.KEYCODE_D);
+//                return true;
+//            }
+//        }
+//        return false;
+//    }
+
+//    @Override
+//    public boolean onInterceptTouchEvent(MotionEvent ev) {
+//        return true;
+//    }
+//    @Override
+//    public boolean onTouchEvent(MotionEvent ev) {
+//        Log.d("Touch", ev.toString());
+//        return true;
+//    }
+//
+//    private void SimulateKeyPress(int simulated) {
+//        new Thread(new Runnable() {
+//            @Override
+//            public void run() {
+//                try {
+////                    Instrumentation inst = new Instrumentation();
+//                    for (int i = 0; i < 10; ++i) {
+//                        inst.sendKeyDownUpSync(simulated);
+//                        Thread.sleep(2000);
+////                        inst.sendKeyDownUpSync(KeyEvent.KEYCODE_BACK);
+////                        Thread.sleep(2000);
+//                    }
+//                }
+//                catch(InterruptedException e) {
+//                }
+//            }
+//        }).start();
+//    }
+//    private void SimulateKeyEvent(KeyEvent keyEvent) {
+//        inst.sendKeySync(keyEvent);
+//    }
+//    private void SimulateKeyDown(int keycode) {
+//        SimulateKeyEvent(new KeyEvent(KeyEvent.ACTION_DOWN, keycode));
+//    }
+//    private void SimulateKeyUp(int keycode) {
+//        SimulateKeyEvent(new KeyEvent(KeyEvent.ACTION_UP, keycode));
+//    }
```

Comment:
Is this comment block a WIP thing for the future, or just a leftover from an experiment that didn't work out?

Reply:
I'm still trying to figure that out. It's looking like it's leaning towards leftover.